File: 9 MaxSliceProblem/MaxDoubleSliceSum/maxDoubleSliceSum.js
function solution(A) {
  const sumsL = A.map(() => 0);
  const sumsR = A.map(() => 0);
  for (let left = 1, right = A.length - 2; right >= 2; left += 1, right -= 1) {
    sumsL[left] = Math.max(0, sumsL[left - 1] + A[left]);
    sumsR[right] = Math.max(0, sumsR[right + 1] + A[right]);
  }
  let max = sumsL[0] + sumsR[2];
  for (let x = 2; x < A.length - 1; x += 1) {
    max = Math.max(max, sumsL[x - 1] + sumsR[x + 1]);
  }
  return max;
}

File: 9 MaxSliceProblem/MaxDoubleSliceSum/maxDoubleSliceSum.cpp
#include <limits>

int solution(vector<int> &A) {
  int sum = 0;
  int output = 0;
  int sum2 = 0;

  for (int i = 3; i < (int)A.size(); i++) {
    sum2 = max(sum2 + A[i - 2], 0);
    sum = max(sum2, sum + A[i - 1]);
    output = max(output, sum);
  }

  return output;
}